webui/react/src/utils/loadable.ts

```diff
@@ -0,0 +1,173 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
```

Comment:
im not very familiar with loadable, but i assume its similar to `Result` type in Rust and Swift. 
idk if we should write our own `Loadable` code. I think it would be used widely, so I think adding library is more reliable and more compatible with React in a long term.
I think we should talk about it before merging it (especially how we handle API calls)
I personally prefer to rely on popular and famous libraries as much as possible. Although this code could be perfect, someone else would change it in the future, and we might get regressions like `useSettings`.

Reply:
It's better not to pull in libraries for things this small, it makes it difficult to audit the code and ensure we don't have security issues which is why I've never made Loadable into a library and just copied it around personal projects.

Reply:
my concern is that if we add global state management tool, this code could be tech debt.
Also, React core team is talking about adding [`use` feature](https://github.com/reactjs/rfcs/pull/229) which is a similar purpose of usage. Even if its still under discussion, next.js v13 already added `use` func, so its likely to be added.

Reply:
You don't have to worry about ADTs becoming tech debt, they're a fundamental programming technique.